client/components/backup-retention-management/index.tsx

```diff
@@ -0,0 +1,239 @@
+import type { RetentionPeriod } from 'calypso/state/rewind/retention/types';
+import './style.scss';
+
+const BackupRetentionManagement: FunctionComponent = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const siteId = useSelector( getSelectedSiteId ) as number;
+
+	// Query dependencies
+	useQueryRewindSize( siteId );
+	useQueryRewindPolicies( siteId );
+	const requestingSize = useSelector( ( state ) => isRequestingRewindSize( state, siteId ) );
+	const requestingPolicies = useSelector( ( state ) =>
+		isRequestingRewindPolicies( state, siteId )
+	);
+	const isFetching = requestingSize || requestingPolicies;
+
+	// Retention period included in customer plan
+	const planRetentionPeriod = useSelector( ( state ) =>
+		getActivityLogVisibleDays( state, siteId )
+	);
+
+	// Retention period set by customer (if any)
+	const customerRetentionPeriod = useSelector( ( state ) =>
+		getBackupRetentionDays( state, siteId )
+	);
+
+	// The retention days option selected by the customer ( or by default )
+	const [ retentionSelected, setRetentionSelected ] = useState( 0 );
+
+	// The retention days that currently applies for this customer.
+	const [ currentRetentionPlan, setCurrentRetentionPlan ] = useState( 0 );
+	useEffect( () => {
+		if ( isFetching ) {
+			return;
+		}
+
+		if ( customerRetentionPeriod ) {
+			setCurrentRetentionPlan( customerRetentionPeriod );
+		} else if ( planRetentionPeriod ) {
+			setCurrentRetentionPlan( planRetentionPeriod );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ customerRetentionPeriod, planRetentionPeriod ] );
+
+	const storageLimitBytes = useSelector( ( state ) =>
+		getRewindBytesAvailable( state, siteId )
+	) as number;
+	const estimatedCurrentSiteSize = useEstimatedCurrentSiteSize();
+
+	const currentSiteSizeText = useStorageText( estimatedCurrentSiteSize );
+	const storageLimitText = useStorageText( storageLimitBytes );
+
+	const retentionOptionsCards: Record< number, RetentionRadioOptionType > = {
+		[ RETENTION_OPTIONS.RETENTION_DAYS_7 ]: usePrepareRetentionOptions(
+			translate( '7 days' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_7,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_7
+		),
+		[ RETENTION_OPTIONS.RETENTION_DAYS_30 ]: usePrepareRetentionOptions(
+			translate( '30 days' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_30,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_30
+		),
+		[ RETENTION_OPTIONS.RETENTION_DAYS_120 ]: usePrepareRetentionOptions(
+			translate( '120 days' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_120,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_120
+		),
+		[ RETENTION_OPTIONS.RETENTION_DAYS_365 ]: usePrepareRetentionOptions(
+			translate( '1 year' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_365,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_365
+		),
+	};
```

Comment:
Do we really need to redeclare those on every render?

Reply:
I think it makes more sense to have it as a component state. We just refactored it.

Addressed in #73303